net/redisclient.go

```diff
@@ -299,7 +299,7 @@ func (r *RedisRingClient) startUpdater(ctx context.Context) {
 		}
 
 		addrs := r.options.AddrUpdater()
-		if !hasAll(addrs, old) {
+		if addrs != nil && !hasAll(addrs, old) {
```

Comment:
Chose `addrs != nil` instead of `len(addrs) != 0` because function may return empty list of redis for other reasons, So in this case those are 2 different cases.